docs/v0.4/getting-started/docker-for-mac.md

```diff
@@ -160,34 +160,101 @@ NAME     LATEST IMAGE
 square   index.docker.io/$DOCKER_ID/square@sha256:ac089ca183368aa831597f94a2dbb462a157ccf7bbe0f3868294e15a24308f68   square.js   <empty>   <empty>   Ready    1m13s
 ```
 
-## create a deployer
+## create a Knative deployer
+
+The Knative runtime is only available on clusters with Istio and Knative installed. Knative deployers run riff workloads using Knative resources which provide auto-scaling (including scale-to-zero) based on HTTP request traffic, and routing.
+
+```sh
+riff knative deployer create knative-square --function-ref square --tail
+```
+
+After the deployer is created, you can see the hostname by listing deployers.
+
+```sh
+riff knative deployer list
+```
+```
+NAME             TYPE       REF      HOST                                 STATUS   AGE
+knative-square   function   square   knative-square.default.example.com   Ready    28s
+```
+
+### invoke the function
+
+Knative configures HTTP routes on the istio-ingressgateway. Requests are routed by hostname.
+
+Look up the nodePort for the ingressgateway. You should a port value like 30195:
+
+```sh
+INGRESS_PORT=$(kubectl get svc istio-ingressgateway --namespace istio-system --output 'jsonpath={.spec.ports[?(@.port==80)].nodePort}')
+echo $INGRESS_PORT
+```
+
+Invoke the function by POSTing to the ingressgateway, passing the hostname and content-type as headers:
 
-Deployers take a built function and deploy it to a riff runtime. There are two runtimes: core and knative. The core runtime is always available, while the knatve runtime is only available on clusters with Istio and Knative installed.
+```sh
+curl http://localhost:$INGRESS_PORT/ -w '\n' \
+-H 'Host: knative-square.default.example.com' \
+-H 'Content-Type: application/json' \
+-d 7
+```
+```
+49
+```
 
+## create a Core deployer
 
-### create a core deployer
+The Core runtime is available on all riff clusters. It deploys riff workloads as "vanilla" Kubernetes deployments and services.
 
 ```sh
-riff core deployer create square --function-ref square --tail
+riff core deployer create k8s-square --function-ref square --tail
 ```
 
-After the deployer is created, you can get the service by listing deployers.
+After the deployer is created, you can lookup the deployment and service names for the function.
 
 ```sh
-riff core deployer list
+kubectl get deployer.core.projectriff.io k8s-square -o custom-columns=NAME:.metadata.name,DEPLOYMENT:.status.deploymentName,SERVICE:.status.serviceName
 ```
+```
+DEPLOYMENT   DEPLOY                SVC
+k8s-square   k8s-square-deployer   k8s-square-deployer
+```
+
+### invoke the function
+
+The ClusterIP service created by a deployer is not reachable from outside the cluster, so it is easier to create a new NodePort service.
 
+```sh
+kubectl expose deployment k8s-square-deployer \
```

Comment:
`expose` creates a new Kubernetes service. `kubectl port-forward` is a better way to create a transient connection to the function. For a permanent access, ingress should be setup for the cluster.

Reply:
ok - with change to port-forward